docs/src/components/ColorSwatch/ColorSwatch.component.tsx

```diff
@@ -0,0 +1,105 @@
+// REACT
+import React, { useState } from 'react';
+// VENDOR
+import styled from 'styled-components';
+import { ColorHelpers } from '../ColorHelpers';
+import { Pill } from '../Pill'
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { Typography } from '../../../../src/Typography';
+// THEME
+import { colors } from '../../../../src/theme';
+
+
+// INTERFACES
+interface ColorSwatchProps {
+  colorHexCode: string;
+  colorName: string;
+}
+
+
+// STYLES
+const StyledColorSwatch = styled.div`
+  border: 0.5rem solid ${props => props.highContrastBorderColor}};
+  display: inline-block;
+  margin: 0 2rem 2rem 0;
+  width: 17rem;
+
+  .color-swatch-pill-container {
+    background: ${props => props.colorHexCode };
+    cursor: pointer;
+    padding: 7rem 1rem 1rem 1.5rem;
+  }
+
+  .color-swatch-text-container {
+    border-top: 0.05rem solid ${props => props.highContrastBorderColor};
+    padding: 1rem;
+  }
+
+  .color-swatch-text-container p {
+    margin: 0;
+    padding: 0.25rem 0 0.25rem 0.5rem;
+  }
+
+  .color-swatch-text-container p:hover {
+    background: ${colors.ash.light};
+    border-radius: 1rem;
+    cursor: pointer;
+    text-decoration: underline;
+  }
+
+  .temporary-copy-confirmation {
+    font-size: 1rem;
+    font-weight: 600;
+    position: fixed;
+    padding: 1rem;
+    top: 2rem;
+    right: 2rem;
+    z-index: 10;
+    color: ${colors.success.base};
+    background: ${colors.white.base};
+    border: 0.1rem solid ${colors.success.base};
+  }
+`;
+
+
+// JSX MARKUP
+export const ColorSwatch = ({
+  colorHexCode,
+  colorName
+}: ColorSwatchProps): React.ReactElement<any> => {
+
+  const highContrastBorderColor = ColorHelpers.getHigherContrastColor(colorHexCode);
+  const colorRgbCode = ColorHelpers.getRgbColor(colorHexCode)
+
+  const [isCopied, setIsCopied] = useState(false);
```

Comment:
Did you know you can type your hooks? Me neither until I did some digging:

```
useState<boolean>(false);
```

Reply:
I didn't, but that's cool. I'll update!